numpy/core/fromnumeric.py

```diff
@@ -1019,6 +1019,41 @@ def argmin(a, axis=None, out=None):
     return _wrapfunc(a, 'argmin', axis=axis, out=out)
 
 
+def argmatch(a, v):
+    """
+    Find indices into `a` where elements in `v` match those in `a`.
+
+    Find the indices into an array `a` whose values match those queried in `v`.
+    Both arrays are first flattened to 1D, and the values in `a` must be unique.
```

Comment:
Can we remove the flattening of `v`? Seems like we can meaningfully preserve shape here, and `searchsorted` already knows how to handle Nd input as its second argument